pyomo/solvers/tests/checks/test_GAMS.py

```diff
@@ -162,5 +165,146 @@ def test_keepfiles_gms(self):
+        with SolverFactory("gams", solver_io="gms") as opt:
+            with redirected_subprocess_call() as output:
+                opt.solve(self.m, tee=False)
+        self.assertFalse(output.getvalue())
+        self._check_logfile(exists=False)
+
+    def test_tee(self):
+        with SolverFactory("gams", solver_io="gms") as opt:
+            with redirected_subprocess_call() as output:
+                opt.solve(self.m, tee=True)
+        self.assertTrue(output.getvalue())
+        self._check_logfile(exists=False)
+
+    def test_logfile(self):
+        with SolverFactory("gams", solver_io="gms") as opt:
+            with redirected_subprocess_call() as output:
+                opt.solve(self.m, logfile=self.logfile)
+        self.assertFalse(output.getvalue())
+        self._check_logfile(exists=True)
+
+    def test_tee_and_logfile(self):
+        with SolverFactory("gams", solver_io="gms") as opt:
+            with redirected_subprocess_call() as output:
+                opt.solve(self.m, logfile=self.logfile, tee=True)
+        self.assertTrue(output.getvalue())
+        self._check_logfile(exists=True, expected=output.getvalue())
+
+
+@unittest.skipIf(not gamspy_available, "The 'gams' python bindings are not available")
+class GAMSLogfilePyTests(GAMSLogfileTestBase):
+    """Test class for testing permultations of tee and logfile options.
+
+    The tests build a simple model and solve it using the different options
+    using the python gams bindings.
+
+    """
+
+    def test_no_tee(self):
+        with SolverFactory("gams", solver_io="python") as opt:
+            with redirected_stdout() as output:
+                opt.solve(self.m, tee=False)
+        self.assertFalse(output.getvalue())
+        self._check_logfile(exists=False)
+
+    def test_tee(self):
+        with SolverFactory("gams", solver_io="python") as opt:
+            with redirected_stdout() as output:
+                opt.solve(self.m, tee=True)
+        self.assertTrue(output.getvalue())
+        self._check_logfile(exists=False)
+
+    def test_logfile(self):
+        with SolverFactory("gams", solver_io="python") as opt:
+            with redirected_stdout() as output:
+                opt.solve(self.m, logfile=self.logfile)
+        self._check_logfile(exists=True)
+        self.assertFalse(output.getvalue())
+
+    def test_tee_and_logfile(self):
+        with SolverFactory("gams", solver_io="python") as opt:
+            with redirected_stdout() as output:
+                opt.solve(self.m, logfile=self.logfile, tee=True)
+        self.assertTrue(output.getvalue())
+        self._check_logfile(exists=True, expected=output.getvalue())
+
+
+@contextlib.contextmanager
+def redirected_stdout():
+    """Temporarily redirect stdout into a string buffer."""
+    output = StringIO()
+    try:
+        setup_redirect(output)
+        yield output
+    finally:
+        reset_redirect()
+
+
+@contextlib.contextmanager
+def redirected_subprocess_call():
```

Comment:
We can delete this method.

Reply:
This can't be deleted as far as I can see since what it does is use the output it gathers and write it into the stream buffer it returns.  Using redirect_stdout unfortunately results in the same empty string we had before.